packages/nextjs/app/debug/_components/contract/utilsDisplay.tsx

```diff
@@ -24,33 +24,87 @@ export const displayTxResult = (
   if (typeof displayContent === "bigint") {
     try {
       const asNumber = Number(displayContent);
-      if (asNumber <= Number.MAX_SAFE_INTEGER && asNumber >= Number.MIN_SAFE_INTEGER) {
-        return asNumber;
+      if (asText) {
+        if (asNumber <= Number.MAX_SAFE_INTEGER && asNumber >= Number.MIN_SAFE_INTEGER) {
+          return asNumber;
+        } else {
+          return "Ξ" + formatEther(displayContent);
+        }
       } else {
-        return "Ξ" + formatEther(displayContent);
+        return <NumberDisplay value={displayContent} />;
       }
     } catch (e) {
       return "Ξ" + formatEther(displayContent);
     }
   }
 
-  if (typeof displayContent === "string" && isAddress(displayContent)) {
-    return asText ? displayContent : <Address address={displayContent} />;
+  if (typeof displayContent === "string") {
+    if (isAddress(displayContent)) {
+      return asText ? displayContent : <Address address={displayContent} />;
+    }
+
+    if (isHex(displayContent)) {
+      return displayContent; // don't add quotes
+    }
   }
 
   if (Array.isArray(displayContent)) {
-    const mostReadable = (v: DisplayContent) =>
-      ["number", "boolean"].includes(typeof v) ? v : displayTxResultAsText(v);
-    const displayable = JSON.stringify(displayContent.map(mostReadable), replacer);
-
-    return asText ? (
-      displayable
-    ) : (
-      <span style={{ overflowWrap: "break-word", width: "100%" }}>{displayable.replaceAll(",", ",\n")}</span>
-    );
+    return asText ? JSON.stringify(displayContent, replacer) : <TupleDisplay value={displayContent} />;
+  }
+
+  if (typeof displayContent === "object") {
```

Comment:
While writing to contract, since we are doing a geneic object check here, the TxReceipt is also showing "struct"  keyword : 

<details>

<summary>
Example : 
</summary>

![Screenshot 2024-05-29 at 3 30 13 PM](https://github.com/scaffold-eth/scaffold-eth-2/assets/80153681/cff802d0-ae67-4204-8357-32d45ef121a0)


</details>

Reply:
Would adding an `isReceipt` variable be fine? We could then refactor out a component, say, `ObjectFieldDisplay` that displays a field (`[key]: [val]`) and we could use that in `TupleDisplay` (where `[key]` is the index), in `StructDisplay` (where `[key]` is the field name), and also in a new `ReceiptDisplay` (where there is no title).

Not sure if that made sense, feel free to ask for clarification.

Reply:
>  say, ObjectFieldDisplay that displays a field ([key]: [val]) and we could use that in TupleDisplay (where [key] is the index), in StructDisplay (where [key] is the field name), and also in a new ReceiptDisplay (where there is no title).

Yeah I like this ! Let's create `ObjectFieldDisplay` component use that in ArrayDisplay and StuctDisplay. 

Regarding `isReceipt`, Lets not complicate `displayTxResult` more, keep as it is. 

I think after having `ObjectFieldDisplay` component, we should be directly use it in `TxReceipt`, so here : 

https://github.com/scaffold-eth/scaffold-eth-2/blob/5df118e637e6d33dc113756ab319052e606de604/packages/nextjs/app/debug/_components/contract/TxReceipt.tsx#L43

```diff
- <pre className="text-xs pt-4">{displayTxResult(txResult)}</pre> 
+ <ObjectFiledDisplay  data={txResult} />

// OR 

<pre className="text-xs pt-4"> 
+ <ObjectFiledDisplay  data={txResult} />
</pre> 

```

Which ever looks good

Reply:
Hmm, just realized the problem with this: `ObjectFieldDisplay` recursively calls `displayTxResult`, so while the first level works, the nested items still show tuple/struct:
![image](https://github.com/scaffold-eth/scaffold-eth-2/assets/104588511/017bcfe9-ed12-40e2-82d2-35d54b08d999)

Should I just leave it like the above picture?

Reply:
I think the picture looks fine, lol bug turned into feature ? 

I think it's not too bad, and also handling it will add more complexity 

So yeah for this PR lets have it similar to pic without "struct" word just below "Transaction Receipt"

Reply:
Alright, committed in https://github.com/ByteAtATime/scaffold-eth-2/commit/2183474b83e46167046e464a467a37be02a66811

I forgot about that hehe, updated in https://github.com/ByteAtATime/scaffold-eth-2/commit/35adf37fbf8a85078d00b40de8c14f56fc3d4d50